posthog/tasks/special_migrations.py

```diff
@@ -0,0 +1,78 @@
+from celery.result import AsyncResult
+
+from posthog.celery import app
+from posthog.special_migrations.runner import (
+    is_current_operation_resumable,
+    run_migration_healthcheck,
+    run_special_migration_next_op,
+    start_special_migration,
+    update_migration_progress,
+)
+from posthog.special_migrations.utils import force_stop_migration, process_error, trigger_migration
+
+
+class CeleryTaskState:
+    Pending = "PENDING"
+    Started = "STARTED"
+    Retry = "RETRY"
+    Failure = "FAILURE"
+    Success = "SUCCESS"
+
+
+# we're hijacking an entire worker to do this - consider:
+# 1. spawning a thread within the worker
+# 2. suggesting users scale celery when running special migrations
+# 3. ...
+@app.task(ignore_result=False, max_retries=0)
```

Comment:
I'd do both 1 & 2 honestly. Threading here would work great since most of this is async waiting on CH do to its thing so no worries with GIL being blocker.
2 in general is probably a good idea (minimum 2 workers always)
3 - the only other implementation I could think of is spinning up an entirely new process using subprocess call or or another process using python3's [multiprocessing](https://docs.python.org/3/library/multiprocessing.html)

Reply:
Will consider threading post-MVP and focus on #2 for now. 

I think it's worth it, just adds some complexity to consider.